### ISSUE TITLE
all composable apps that coerce data to a moltype should preserve class

### ISSUE THREAD

**Issue by GavinHuttley** [cogent3/cogent3#243]
use the `to_moltype()` method, subject of #242

**Comment by GavinHuttley** [comment]
There are multiple cases of this, only in `cogent3.app`. Here's a specific example in `cogent3.app.sample`
```python
    def filter_degenerates(self, aln):
        if aln.moltype != self.moltype:
            # try converting
            array_align = isinstance(aln, ArrayAlignment)
            aln = aln.to_type(moltype=self.moltype, array_align=array_align)

```
should be changed to
```python
    def filter_degenerates(self, aln):
        if aln.moltype != self.moltype:
            aln = aln.to_moltype(self.moltype)
```
Suggest just searching for `to_type` in the `app` files

**Comment by GavinHuttley** [comment]
fixed in b89da485c30d68442a5c46cc65c94f1c73f7ed70
